front/src/modules/rollingStock/components/RollingStockSelector/RollingStockSelector.tsx

```diff
@@ -1,35 +1,56 @@
-import React, { useRef } from 'react';
+import React, { useEffect, useRef, useState } from 'react';
 import RollingStockModal from 'modules/rollingStock/components/RollingStockSelector/RollingStockModal';
+import RollingStock2Img from 'modules/rollingStock/components/RollingStock2Img';
 import icon from 'assets/pictures/components/train.svg';
 import {
   comfort2pictogram,
   RollingStockInfo,
 } from 'modules/rollingStock/components/RollingStockSelector/RollingStockHelpers';
 import { useModal } from 'common/BootstrapSNCF/ModalSNCF';
-import { RollingStockComfortType, RollingStock } from 'common/api/osrdEditoastApi';
+import { RollingStockWithLiveries, osrdEditoastApi } from 'common/api/osrdEditoastApi';
+import { useTranslation } from 'react-i18next';
+import { useOsrdConfSelectors } from 'common/osrdContext';
+import { useSelector } from 'react-redux';
 
 type RollingStockProps = {
-  rollingStockSelected?: RollingStock;
-  rollingStockComfort?: RollingStockComfortType;
-  image?: JSX.Element;
-  comfort?: string;
-  comfortType?: string;
-  choice?: string;
   condensed?: boolean;
 };
 
-const RollingStockSelector = ({
-  rollingStockSelected,
-  rollingStockComfort,
-  image,
-  comfort,
-  comfortType,
-  choice,
-  condensed,
-}: RollingStockProps) => {
+const RollingStockSelector = ({ condensed }: RollingStockProps) => {
   const { openModal } = useModal();
 
   const ref2scroll = useRef<HTMLDivElement>(null);
+
+  const [rollingStockSelected, setRollingStockSelected] = useState<
+    RollingStockWithLiveries | undefined
+  >();
+
+  const { t } = useTranslation(['translation', 'rollingstock']);
+  const { getRollingStockID, getRollingStockComfort } = useOsrdConfSelectors();
+  const rollingStockId = useSelector(getRollingStockID);
+  const rollingStockComfort = useSelector(getRollingStockComfort);
+  const comfort = t('rollingstock:comfort');
+  const comfortType = t(`rollingstock:comfortTypes.${rollingStockComfort}`);
+  const choice = t('rollingstock:rollingstockChoice');
+
+  const [getRollingStockByID] = osrdEditoastApi.useLazyGetRollingStockByRollingStockIdQuery();
+
+  const getRollingStock = () => {
+    if (rollingStockId) {
+      getRollingStockByID({ rollingStockId })
+        .then(({ data }) => setRollingStockSelected(data))
+        .catch((err) => console.error(err));
+    }
+  };
+
+  useEffect(() => {
+    if (rollingStockId !== undefined) {
+      getRollingStock();
+    }
+  }, [rollingStockId]);
```

Comment:
You don't need to do all that stuff. You can use the regular useQuery from rtk query and then remove the state
```suggestion
 const { data: rollingStockSelected } =
    osrdEditoastApi.endpoints.getRollingStockByRollingStockId.useQuery(
      {
        rollingStockId: rollingStockId as number,
      },
      {
        skip: !rollingStockId,
      }
    );
```

Reply:
The rollingStock is already fetched in the parent component. In order to try to isolate the current component, we should pass the rs through the props. You can also pass the rollingStock comfort, this way we remove the selector from the component (and it is fully isolated :) )